### ISSUE TITLE
Skip to content functionality

### ISSUE THREAD

**Issue by joelit** [NatLibFi/Skosmos#1048]
When browsing the Skosmos pages with either vocabulary list, vocabulary info, search results or concept info, the browser focus should go first to the skip to content -link in:
https://github.com/NatLibFi/Skosmos/blob/d633b60c95b39cf9c7fe261fca03097b3e057d1e/view/light.twig#L23

This does not happen in practise, as the focus is moved to the search field first. The culprit seems to be docready.js, which alters the focus in various ways, e.g.

https://github.com/NatLibFi/Skosmos/blob/d633b60c95b39cf9c7fe261fca03097b3e057d1e/resource/js/docready.js#L1109

This issue was split from #1036 #1037 #1038 & #1039

**Comment by osma** [comment]
Related discussion on SO: https://stackoverflow.com/questions/2180645/is-automatically-assigning-focus-bad-for-accessibility

**Comment by osma** [comment]
The more modern way to implement autofocus would be to use the `autofocus` attribute in HTML5 instead of JavaScript. But regardless of the autofocus implementation, it causes accessibility concerns. 

My current understanding is that autofocus should be used very sparingly - only on pages whose primary purpose is text input (e.g. Google homepage style search, login forms, contact forms).

I propose dropping the autofocus JS code from the `docready.js` script in Skosmos and, if necessary, using the `autofocus` attribute on pages that really deserve it - I think only the contact form qualifies. 

Some more background reading:
* https://www.brucelawson.co.uk/2009/the-accessibility-of-html-5-autofocus/
* https://www.samanthaming.com/tidbits/42-html5-autofocus/#accessibility-concerns
* https://github.com/whatwg/html/issues/4575

**Comment by joelit** [comment]
Thanks, I'll give it a read-through!

I was thinking of handling the focus order through HTML, as I don't see a reason why the tabindex wouldn't be enough for the search field.

**Comment by osma** [comment]
Tested the "skip to content" functionality in this PR on a text-based browser (lynx).

* Front page: works as expected, takes me to the beginning of the vocabulary list
* Vocabulary page: not what I expected - takes me to the A-Z index in the sidebar, not the main content (which would be the information about the vocabulary)
* Search result page: not what I expected - takes me to the search limit form, not the search result list
* Concept page: not what I expected - takes me to the A-Z index in the sidebar, not the main content (which would be the information about the concept)

I think the `main` element should be moved deeper/later into the DOM tree on all page types except the front page - after the sidebar (which is a navigational element, not part of main content). Otherwise the user will not land in the right place when selecting this link.
